### PR TITLE
chore: use eslint + prettier

### LINKED ISSUE #451: Update dev tooling
* Replace tslint with eslint
* Use https://github.com/vega/vega-lite-dev-config

### PR DESCRIPTION
**Pull request by leibatt**
Please:
- [x] Make your pull request atomic, fixing one issue at a time unless there are many relevant issues that cannot be decoupled.
- [ ] Provide a test case
- [x] Make lint and test pass. (Run `npm run lint` and `npm run test`)
- [x] Make sure you have merged `master` into your branch.
- [x] Provide a concise title so that we can just copy it to our release note.
  - Use imperative mood and present tense.
  - Mention relevant issues. (e.g., `#1`)

Should fix #451

Steps:
* added vega-lite-dev-config
* created beemo config files
* updated `package.json` scripts to use eslint and prettier
* ran `yarn test` to see what failed
* ran `yarn format` to fix the easy stuff
* modified rules levels in `.config/beemo/eslint.ts` to warn for most remaining linting errors
* fixed very minor errors in a couple of files:
  * `src/util.ts`
  * `test/constraint/field.test.ts`
  * `index.html`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+module.exports = {
```
Comment (domoritz) [inline .eslintrc.js:1]: This file should be in the gitignore since beeil generates it.

File: prettier.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (domoritz) [inline prettier.config.js:1]: This file should be in the gitignore since it's auto generated.

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.eslintrc.js
```
Comment (domoritz) [inline .prettierignore:1]: Can be ignored since it's auto generated

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.eslintrc.js
```
Comment (domoritz) [inline .eslintignore:1]: Can be ignored since it's auto generated

**Review by domoritz — APPROVED** [review]
Looks great. Let's remove the auto-generated files and we are good to merge.

File: .eslintrc.js (hunk 1 above)
Comment (leibatt) [reply .eslintrc.js]: added to .gitignore

File: prettier.config.js (hunk 2 above)
Comment (leibatt) [reply prettier.config.js]: added to .gitignore

File: .prettierignore (hunk 3 above)
Comment (leibatt) [reply .prettierignore]: added to .gitignore

File: .eslintignore (hunk 4 above)
Comment (leibatt) [reply .eslintignore]: added to .gitignoore

**Pull request merged**